cli/src/commands/dsl/validate.ts

```diff
@@ -34,6 +35,77 @@ export default class Validate extends BaseCommand {
     };
   }
 
+  private async extendTSConfig(filePath: string, compilerOptions: object) {
+    let extConfigFile, extTSEnvConfig;
+
+    const extendedTSConfig = glob.sync(filePath)[0];
+
+    this.log(
+      `Local extended Typesscript config file found ${extendedTSConfig}`
+    );
+
+    try {
+      extConfigFile = await fs.readFile(extendedTSConfig);
+    } catch (e) {
+      this.log(
+        "Error reading the TypeScript extended configuration file. Using tsconfig compile options only"
+      );
+      return compilerOptions;
+    }
+
+    const extCompilerConfigObject =
+      extConfigFile && JSON.parse(extConfigFile.toString());
+
+    if (extCompilerConfigObject.compilerOptions) {
+      extTSEnvConfig = extCompilerConfigObject.compilerOptions;
+
+      return { ...compilerOptions, ...extTSEnvConfig };
+    } else {
+      this.log("No extended Typescript Compiler options found in file");
+    }
+
+    return compilerOptions;
+  }
+
+  private async getTSConfig() {
+    let TSEnvConfig, configFile;
+
+    const EnvTSConfig = glob.sync("./tsconfig.json")[0];
+
+    this.log(`Local Typesscript config file found ${EnvTSConfig}`);
+
+    try {
+      configFile = await fs.readFile(EnvTSConfig);
+    } catch (e) {
+      this.log("Error reading the TypeScript configuration file.");
+      return;
+    }
+
+    const compilerConfigObject =
+      configFile && JSON.parse(configFile.toString());
+
+    if (compilerConfigObject.compilerOptions) {
+      TSEnvConfig = compilerConfigObject.extends
+        ? await this.extendTSConfig(
```

Comment:
Can we do this in a recursive way so that we don't need two separate functions for handling the base config and an extended config. That way if there is another level of extended config its automatically supported.